GitUp/Application/AppDelegate.m

```diff
@@ -69,6 +69,38 @@ - (AboutWindowController *)aboutWindowController {
   return _aboutWindowController;
 }
 
+- (WelcomeWindowController *)welcomeWindowController {
+  if (!_welcomeWindowController) {
+    _welcomeWindowController = [[WelcomeWindowController alloc] init];
+    
+    __weak NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+    _welcomeWindowController.model.getUserDefaultsShouldShow = ^BOOL{
+      return [defaults boolForKey:kUserDefaultsKey_ShowWelcomeWindow];
+    };
+    
+    __weak NSDocumentController *documentController = NSDocumentController.sharedDocumentController;
+    _welcomeWindowController.model.getRecentDocuments = ^NSArray<NSURL *> * _Nonnull{
+      return documentController.recentDocumentURLs;
+    };
```

Comment:
If it's not explicitly tied to the `AppDelegate` (NSUserDefaults have `standardUserDefaults`, NSDocumentController has a `sharedDocumentController`), I'm not really convinced these blocks make the code clearer. I'd prefer the window to handle this logic rather than trying to inject from the `AppDelegate` level.